core/metamodel/src/main/java/org/apache/isis/core/metamodel/services/ServicesInjector.java

```diff
@@ -78,7 +82,7 @@
      * services that are assignable to the type.  It's possible that this is an empty list.
      */
     private final Map<Class<?>, List<Object>> servicesAssignableToType = _Maps.newHashMap();
-    private final Map<Class<?>, Object> serviceByConcreteType = _Maps.newHashMap();
+    private final _Lazy<Map<Class<?>, Object>> serviceByConcreteType = _Lazy.of(this::initServiceByConcreteType);
```

Comment:
internal API: _Lazy
this is a perfect showoff on how easy it is to replace a final field with a lazy 'wrapper'
I've slightly adopted this idea from 'vavr' former 'java-slang'

(there's also a thread-safe variant to this)

Reply:
Very elegant!